### ISSUE TITLE
Set up tests for supported permissions

### ISSUE THREAD

**Issue by nickoneill** [nickoneill/PermissionScope#45]
I haven't played with the UI testing in Xcode 7 yet but I expect we'll need lots of UI tests since permissions require user input. Hopefully we can automate this.

My preference would be something like this:
- [ ] Run a set of UI tests that create a few dialogs with each of the permissions
- \- [ ] Verify all permissions can be denied and detected as such
- \- [ ] Verify all permissions can be approved and detected as such
- [ ] Any tests that require devices (bluetooth, for one) can be hooked up to a device and run similarly (denied, delete/reset and then approved)

Resetting the simulator gives a fresh slate for permissions, unlike just deleting an app from a device. We'll have to figure out how to balance this with permissions that require a device.

**Comment by felix-dumit** [comment]
The hackish way to re-request permissions on the device is to change the bundle id, we'd probably need to make a script to run the tests and also change the bundle in that case.

**Comment by nickoneill** [comment]
Good idea, that's an excellent way of testing on device/sim without having to reset content and settings :/

**Comment by bre7** [comment]
Sadly UIT doesn't support interaction with system alerts as far as I know...It'll be probably added by the time GM is released.

**Comment by nickoneill** [comment]
Ugh. I'll do some testing when Xcode 7b3 comes out and file some radars if we can't.

**Comment by bre7** [comment]
I think you can interact with system alerts now. The recording feature does detect them at least.

I've found something in the forums related to a UI testing (iOS9+) bug when interacting with system alerts: https://forums.developer.apple.com/thread/11574 but not sure if it stills hapenning.
